pkg/scanner/scan.go

```diff
@@ -34,35 +32,76 @@ func AddAcceptanceExtensions(keys []string) {
 	}
 }
 
-func ScanImage(ctx context.Context, imageName, filePath string, dockerOption deckodertypes.DockerOption) (assessments []*types.Assessment, err error) {
-	var ext extractor.Extractor
-	var cleanup func()
-	if imageName != "" {
-		ext, cleanup, err = docker.NewDockerExtractor(ctx, imageName, dockerOption)
-		if err != nil {
-			return nil, fmt.Errorf("%v. %w", err, types.ErrorCreateDockerExtractor)
-		}
-	} else if filePath != "" {
-		ext, cleanup, err = docker.NewDockerArchiveExtractor(ctx, filePath, dockerOption)
-		if err != nil {
-			return nil, fmt.Errorf("%v. %w", err, types.ErrorCreateDockerExtractor)
-		}
+func ScanImage(ctx context.Context, cfg config.Config) ([]*types.Assessment, error) {
+	registryOptions := image.RegistryOptions{
+		InsecureSkipTLSVerify: config.Conf.Insecure,
+		Credentials: []image.RegistryCredentials{
+			{
+				Username: config.Conf.Username,
+				Password: config.Conf.Password,
+				Token:    config.Conf.Token,
+			},
+		},
+		InsecureUseHTTP: config.Conf.NonSSL,
+	}
+
+	var userInput string
+	if cfg.FilePath != "" {
+		userInput = cfg.FilePath
+	} else if cfg.ImageName != "" {
+		userInput = setImageSource(cfg.LocalImage, cfg.ImageName)
 	} else {
 		return nil, types.ErrSetImageOrFile
 	}
-	defer cleanup()
-	ac := analyzer.New(ext)
-	var files deckodertypes.FileMap
+
+	opts := stereoscope.WithRegistryOptions(registryOptions)
+	img, err := stereoscope.GetImage(ctx, userInput, opts)
```

Comment:
you are missing `image.Cleanup()`

```
// Cleanup removes all temporary files created from parsing the image. Future calls to image will not function correctly after this call.
```

not sure when you should call this cleanup flow so it will not delete stuff that you need.

Reply:
I've updated.

Reply:
@pbalogh-sa you are storing the `img` on 
```
imageData := &types.ImageData{
		Image:   img,
		FileMap: files,
	}
```
I'm not sure it is safe to use the defer here.

Reply:
after discussing with @pbalogh-sa we made sure that `img` is not being used outside of the flow of this function so it's safe to clean it here.